model/riscv_vmem_sv32.sail

```diff
@@ -141,7 +141,7 @@ function translate32(asid, ptb, vAddr, ac, priv, mxr, do_sum, level, ext_ptw) =
                 n_ent.pte = n_pte.bits();
                 write_TLB32(idx, n_ent);
                 /* update page table */
-                match mem_write_value(to_phys_addr(EXTZ(ent.pteAddr)), 4, n_pte.bits(), false, false, false) {
+                match mem_write_value_priv(to_phys_addr(EXTZ(ent.pteAddr)), 4, n_pte.bits(), default_write_acc, Supervisor, false, false, false) {
```

Comment:
Hm, the fact that `mem_write_value_priv` is always called with default_write_acc is probably a sign that parameter should be dropped, like how it doesn't exist for `mem_write_value`? Then `mem_write_value` should just call `mem_write_value_meta` directly (rather than constructing `Write(default_write_acc)` to call `effectivePrivilege`, throwing the access type away and `mem_write_value_priv` re-creating the same thing.

Reply:
I struggled for a bit with the right shape of things here.  Are you suggesting, then,

```
/* Explicit AccessType, Privilege, and metadata */
val mem_write_value_meta : forall 'n, 0 < 'n <= max_mem_access . (xlenbits, atom('n), bits(8 * 'n), AccessType(ext_access_type), Privilege, mem_meta, bool, bool, bool) -> MemoryOpResult(bool) effect {wmv, wmvt, rreg, wreg, escape}

/* Explicit Privilege, default AccessType and metadata */
val mem_write_value_priv : forall 'n, 0 < 'n <= max_mem_access . (xlenbits, atom('n), bits(8 * 'n), Privilege, bool, bool, bool) -> MemoryOpResult(bool) effect {wmv, wmvt, rreg, wreg, escape}
function mem_write_value_priv (paddr, width, value, priv, aq, rl, con) =
￼  mem_write_value_meta(paddr, width, value, Write(default_write_acc), priv, default_meta, aq, rl, con)

/* Default AccessType, Privilege, and metadata */
function mem_write_value (paddr, width, value, aq, rl, con) = {
￼  let typ = Write(default_write_acc);
￼  let ep = effectivePrivilege(typ, mstatus, cur_privilege);
￼  mem_write_value_meta(paddr, width, value, typ, ep, default_meta, aq, rl, con)
￼}
```

Reply:
Yeah, that's what I was thinking